src/PluginsManager/Microsoft.Performance.Toolkit.Plugins.Core/Transport/IPluginFetcher.cs

```diff
@@ -13,26 +14,33 @@ namespace Microsoft.Performance.Toolkit.Plugins.Core.Transport
     ///     Represents a fetcher that is capable of fetching the plugin package from a supported <see cref="AvailablePluginInfo"/>.
     ///     A fetcher is meant to be stateless.
     /// </summary>
-    public interface IPluginFetcher : IPluginManagerResource
+    public interface IPluginFetcher
+        : IPluginManagerResource
     {
         /// <summary>
         ///     Checks if the given <paramref name="pluginInfo"/> is supported by this fetcher.
         /// </summary>
         /// <param name="pluginInfo">
         ///     The available plugin info containing information for the fetcher to verify if it's supported.
         /// </param>
+        /// <param name="logger">
+        ///     Used to log information.
+        /// </param>
         /// <returns>
         ///     <c>true</c> if <paramref name="pluginInfo"/> is supported by this fetcher. <c>false</c>
         ///     otherwise.
         /// </returns>
-        Task<bool> IsSupportedAsync(AvailablePluginInfo pluginInfo);
+        Task<bool> IsSupportedAsync(AvailablePluginInfo pluginInfo, ILogger logger);
```

Comment:
I am not a fan of the discoverer having a `SetLogger` method, but the fetcher getting them passed through method parameters

Reply:
Trying to keep fetcher thread safe by not adding properties to it (assuming logger is not thread safe?) , as fetcher would be a singleton for plugins manager and can be accessed by multiple threads.  But now that I think about it, I should do the same for `IPluginDiscovererProvider` as all the resources are on the same boat. 

I'm leaning towards `SetLogger` for all of them and use locks in the callers. Any thoughts?

Reply:
Yes, `SetLogger` makes more sense. I would assume it's up to the clients to make sure loggers aren't being concurrently set with calls to fetch/discover things.

Reply:
I've updated to use `SetLogger()`. However, the thing I was worried about is not only the Set itself but also the thread safety of other public methods on `IPluginFetcher` and `IPluginDiscoverer` because they are shared resources and the logger (the default one uses a `TextWriter` under the hood which is not thread safe). Wonder if I would need to ensure thread safety in the callers of all of these methods.